### ISSUE TITLE
Automatic SSL certificate generation

### ISSUE THREAD

**Issue by Neraverin** [felddy/foundryvtt-docker#577]
### Feature description

Add the ability not to transfer the SSL certificate and key pair outside, but to generate (and update) them inside the Docker image

### Motivation

It is impossible to play D&D without communication between the players and the master. FoundryVTT uses WebRTC for audio/video (A/V) conferencing. But this feature requires the obligatory presence of an SSL connection. Many users around the world use Let's Encrypt (certbot) to create and renew SSL certificates. It would be very convenient to be able to automatically start certbot inside the image.

### Example

When building a Docker image, you can use an image that already has nginx and certbot (https://github.com/JonasAlfredsson/docker-nginx-certbot/).
To use the feature, the user will have to set 1 obligatory parameter - his domain name. All other parameters can be optional (enable automatic update, frequency of automatic update). The certificate itself and the key pair can be stored in source: <your_data_dir>.

### Pitch

Now, in order to start playing with your friends, even if you have your own server, you need to:
- install and configure nginx as a proxy
- install and configure certbot
- run docker image
- figure out why nothing works :)

Even the developers of Foundry VTT themselves note that setting up this feature requires additional technical skills. And even if you went all the way successfully, the resulting solution is not the Docker way.

At the same time, if the feature is implemented, all you need is a server with a public IP + buying a domain name (you can use a dynamic DNS service).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Snogard** [comment]
isn't it better to use a dedicate image like [this](https://github.com/linuxserver/docker-swag)?

**Comment by ForrestAM** [comment]
I generally thing it should be separate as foundry itself is only the VTT, and shouldn't really be a whole webserver like that.
Why not just setup Caddy in front or even use traefik? Caddy is dead simple, and Traefik is useful just to learn for docker hosting

**Comment by felddy** [comment]
I agree with @ForrestAM.  This container's raison d'être is to get a containerized FoundryVTT instance configured and running.  Certificate management is non-trivial and there are already very good solutions available that will do it better than I could.  Placing this container behind `caddy` (or a similar proxy) is The Way™. 

I have a cookbook branch that I started working on that has a good example of how to do this with `caddy`.  Note that it is a work-in-progress and I just noticed the diagram has a bunch of `LiveKit` stuff in it that should be ignored:

https://github.com/felddy/foundryvtt-docker/blob/improvement/cookbooks/docs/cookbooks/caddy/README.md
